### PR TITLE
Removes certain warden locker and contraband locker spawns

### PR DESCRIPTION
**Pull request by senjx**
## About The Pull Request

This PR is a follow up to #50746. It retains the availability and functionality of the compact combat shotgun but removes it from the spawn table on the warden locker. Additionally, this removes the weapon spawns from the contraband locker which will now spawn various varieties of donuts.

## Why It's Good For The Game

"Super alpha strike compact shotgun is not really healthy for game balance." - Rohesie
The contraband locker has been subject to a lot of debate and controversy over the past and the previous removal PR showed that maintainers wanted the spawner removed, which I've done by replacing it with a funnier and tamer loot list.

## Changelog
:cl:angelstarri
del: MA Arms Inc. has initiated a product recall of all compact combat shotguns due to faulty internal magazines that cause the weapon to explode after firing. Unfortunately, all of the weapons in our stock have already exploded. We're not going to acquire any replacements.

balance: Nanotrasen Internal Security has cracked down on security personnel rummaging confiscated contraband items originally intended for safe storage. Rumor is that security is now using the locker to store high-value food items.
/:cl:

### CODE REVIEW

**Comment by BeeSting12** [comment]
leave my deagle alone, also ided

**Comment by senjx** [comment]
is everything an ided pr to you beesting

**Comment by actioninja** [comment]
I don't see how this comes across as anything besides ided especially considering the person making it.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 1]
```diff
@@ -46,29 +46,19 @@
 
 /obj/effect/spawner/lootdrop/armory_contraband
 	name = "armory contraband gun spawner"
```
Comment (Fikou) [inline code/game/objects/effects/spawners/lootdrop.dm:48]: gun

**Comment by uomo91** [comment]
If it...
>removes it from the spawn table on the warden locker.

Exactly how does it...?
>retains the availability and functionality of the compact combat shotgun

How else is it otherwise "available"?

Including the contraband locker stuff this time around seems like a thin attempt to repackage this as something other than removing an item you don't like because it makes tiding into the brig/armory harder, which is its intended purpose.

The warden's stated job is to shoot on sight people trespassing in the armory (a capital offense in both RP and the server rules), and to handle and sentence dangerous prisoners. They have purview over a room filled with _other_ lethal firearms and a department potentially filled with dangerous criminals others might want to break out so I have no idea why giving them a special purpose lethal firearm to deal with all that is somehow bad.

**Comment by Fikou** [comment]
EPIC fix:
turn the useless firing range firing pin into the brig firing pin and install it in the compact shotgun so warden only can shoot it in the brig

**Comment by senjx** [comment]
> How else is it otherwise "available"?

It's available in the code if admins want it. I would've rather liked a complete removal, but I was advised against it in the previous PR which you are more than welcome to read.

>The warden's stated job is to shoot on sight people trespassing in the armory (a capital offense in both RP and the server rules) and to handle and sentence dangerous prisoners. 

I don't know where this mentality of 'kill anyone who dares go against security' comes from but it's getting old really fast. The cloning system removal was supposed to step up the scales and to make Security opt for using other methods of imprisonment other than outright round removal but nowadays it feels like everyone is keen on shooting/killing players first and asking questions second, if not outright hiding/leaving bodies to rot without asking questions at all. 

If you're proposing that Security just kill everyone who commits anything beyond a minor crime, then you can go ahead and create a PR to remove everything but the execution room. The compact shotgun is the enabler of the offenders who seek round removal for those who want to NOT ONLY enter a hazardous or dangerous area but for even entering the fucking brig. I've seen too many shitter warden players who will shoot on sight as soon as someone walks past the foyer. This is evidence enough to warrant just outright remove a weapon that has so much kill potential that is used more for malice than good.

With a wide variety of non-lethal and less than lethal equipment **available at round start** such as:
- Krav Maga
- Beanbag slugs and rubber shot
- Flashes
- Flashbangs
- Stingbangs
- Stun batons
- Disabler beams
- Bolas
- Tear gas
- N2O canisters
- Pepperspray
- DragNETs
 
And many more that you can obtain later in the round. The compact combat shotgun, I repeat, is nothing short of a grief enabler for Security players who want to do more harm than good. 

> Including the contraband locker stuff this time around seems like a thin attempt to repackage this as something other than removing an item you don't like because it makes tiding into the brig/armory harder, which is its intended purpose.

I'm killing two birds with one stone. The contraband locker has had too much controversy in the past due to its existence and I'm the one who's stepping up to remove it. If you're insinuating I made this PR because I want to get an advantage for whenever I'm playing assistant, then you can kindly fuck off. You can say that to just about any removal PR and has no leverage on any worthwhile argument.

> The warden's stated job is to shoot on sight people trespassing in the armory (a capital offense in both RP and the server rules),

Since you're making this argument based on the Manuel ruleset, let me remind you of this cute little paragraph that 0% of the player base knows about:

`Only the Captain, HoS, and Warden can authorize a Permanent Sentence.
Only the Captain can authorize an Execution or Forced Cyborgization.`

Nowhere in the server rules does it say that the warden's job is to outright kill people for trespassing into a dangerous area. This is just completely and utterly neo-player attitude garbage of having to do the less possible effort to stop someone by killing them in less than 2 seconds with a pocket combat shotgun, that not only makes combat dull but makes every situation on combat against someone with it incredibly predictable.

I'm open to seeing a citation for your statement of this being in the server rules if you're willing to provide one. Until then, I'll err on the side of not enabling players who seek to play Security only to kill people with the most powerful shotgun in the game. Wardens already get a starting laser gun on their locker apart from the normal locker inventory. It's time to remove some of the extra aces in the hole Warden players have.

**Comment by senjx** [comment]
> EPIC fix:
> turn the useless firing range firing pin into the brig firing pin and install it in the compact shotgun so warden only can shoot it in the brig

bloat solution that only works for one item

**Comment by uomo91** [comment]
>wall of angry text

Well I guess you've proved me wrong that you're not emotionally invested in this for reasons relating to your play style, lol.

>Nowhere in the server rules does it say that the warden's job is to outright kill people for trespassing into a dangerous area.

If by "nowhere in the server rules" you mean "literally in the server rules pretty plainly" then you are 100% correct.

https://tgstation13.org/wiki/Rules#Security_Policy_.26_Precedents

>6. ... lethal force may be used immediately on anyone trying to enter the armory, is in the armory, or is leaving it

**Comment by senjx** [comment]
I was going to reply seriously until I remembered you're one of those wardens who kill people for the smallest offense. I'll stop taking the bait now. Go work on your irrelevant sign fix instead.

**Comment by tmtmtl30** [comment]
look, there are two possibilities for people breaking into the armory:

-they're greytiding shitters who deserve the bullet
-they're antagonists looking to murderbone who deserve the bullet

i see no problem with lethal force

**Comment by BeeSting12** [comment]
I can respect that you're willing to write five paragraphs on why the warden shouldn't be killing people who are trying to gain access to lethal weapons but I disagree and I will not be writing a five paragraph response.

**Comment by Shadowflame909** [comment]
>Removing the hideable portable shotgun
Very Based
>Removing the contraband maint spawner
Not fun

The chance of random loot to shake up the game is a staple of ss13! Every department gets one, whether it be strange seeds, or an xenomorph egg, or a wizard d20 in maint.

Just replace it with less lethal items. A good fit would be the giant sized e-bow. Would fit right in as a replacement for the deagle. Not lethal, same level of illegal tech being unlocked. Very rewarding in terms of RNG.

If not that, how about a bananium e-sword/holographic e-sword.

**Comment by tmtmtl30** [comment]
the compact combat shotgun isn't hideable?? it's literally a bulky item, you can't put it in your backpack. you CAN put it in your suit storage slot, but the suit storage slot is visible on examine. you can tell if people have it or not by looking closely.

**Comment by uomo91** [comment]
> the compact combat shotgun isn't hideable?? it's literally a bulky item, you can't put it in your backpack. you CAN put it in your suit storage slot, but the suit storage slot is visible on examine. you can tell if people have it or not by looking closely.

And it can be stripped and stolen as quickly and easily as any other worn item.

**Comment by Shadowflame909** [comment]
View the previous thread on this topic to see why I truly dont like shotguns that can hold more then one bullet, that you can store away anywhere

**Comment by SomeoneYouProbablyKnow** [comment]
A) I'm unsure those 2 changes are about the same problem and i'm sure there is a less blunt solution to the contraband locker (if there is a problem even)

B) You wanna remove the shotgun from the locker? Replace it with a riot shotgun (or a combat shotgun) not to reduce sec firepower too much for stuff like nukies cult etc. Wardens will take one from the armoury anyway

**Comment by Pugie** [comment]
You seem pretty passionate about this. It's just a unique shotgun with a 5 shell capacity and has a 0.2 second less fire delay compared to riot shotgun when it comes to shooting. It's a minor upgrade, hell maybe it's even closer to a side-grade considering the lower capacity. 

Don't see how this is any more of a "shitsec enabler" over say a laser gun or riot shotgun which will also shoot you dead when used lethally, your issues seem to be more up the security policy alley and stripping nice unique jobstuffs saddens me.

**Comment by Fikou** [comment]
> You seem pretty passionate about this. It's just a unique shotgun with a 5 shell capacity and has a 0.2 second less fire delay compared to riot shotgun when it comes to shooting. It's a minor upgrade, hell maybe it's even closer to a side-grade considering the lower capacity.

isnt the entire point of combat shotgun that they are semi automatic? you dont need to pump them after shooting unlike riot shotguns

**Comment by Pugie** [comment]
> > You seem pretty passionate about this. It's just a unique shotgun with a 5 shell capacity and has a 0.2 second less fire delay compared to riot shotgun when it comes to shooting. It's a minor upgrade, hell maybe it's even closer to a side-grade considering the lower capacity.
> 
> isnt the entire point of combat shotgun that they are semi automatic? you dont need to pump them after shooting unlike riot shotguns

have you tried pressing the z key very fast?

**Comment by tmtmtl30** [comment]
eh, it's still an impediment, especially if you forget to do it until you're in combat or alternatively if you do it too early and lose a shell. not a major thing, enough that i sometimes take the riot shottie over the compact combat shotgun because of the increased capacity (and also a little because i often eject the next round when using the combat shotgun off of muscle memory), but it's there

**Comment by bobbahbrown** [comment]
> Since you're making this argument based on the Manuel ruleset, let me remind you of this cute little paragraph that 0% of the player base knows about:
> 
> `Only the Captain, HoS, and Warden can authorize a Permanent Sentence. Only the Captain can authorize an Execution or Forced Cyborgization.`
> 
> Nowhere in the server rules does it say that the warden's job is to outright kill people for trespassing into a dangerous area. This is just completely and utterly neo-player attitude garbage of having to do the least possible effort to stop someone by killing them in less than 2 seconds with a pocket combat shotgun, that not only makes combat dull but makes every situation on combat against someone with it incredibly predictable.

Hi Senjie,

I would like to remind you that the line you are quoting is not a rule, not policy, and in fact is an in-character roleplay recommendation. I believe this should be clarified as what you are saying is disingenuous.

What you quoted is from the [space law page](https://tgstation13.org/wiki/Space_Law), and I would like to remind you of this large text at the top of the page.
>If you cite Space Law in an adminhelp or anywhere outside IC communications, you will be laughed at.

Best,
bobbah "bee" brown

**Comment by bobbahbrown** [comment]
Additionally

>The cloning system removal was supposed to step up the scales and to make Security opt for using other methods of imprisonment other than outright round removal...

> The contraband locker has had too much controversy in the past due to its existence and I'm the one who's stepping up to remove it.

Could you provide more detail into either of the above statements? I haven't seen anyone talking about the above two things and would like to look into it more.

> I'm open to seeing a citation for your statement of this being in the server rules if you're willing to provide one...

Why did you disregard the genuine citation provided to you afterwards, and instead personally attack the person who provided it to you? Are you uncomfortable following up on their citation?

**Comment by SomeoneYouProbablyKnow** [comment]
I think there have been a minor controversy with the contraband locker giving illegal tech (and the advanced cam console) basically roundstart. But as i said before there should be a more subtle solution to that problem (IF you consider it a problem)

**Comment by Tlaltecuhtli** [comment]
illegal tech roundstart is a misconception that cant be done as even if you unlock the node you need industrial tech.weapon tech, adv weapon tech, grenades and then illegal tech with each node costing 10k, you need 42500 research points and then you are cockblocked by minerals as once you get the first diamonds you wont have enough to print a single emag module as you have no t3/t4 to make shit not cost 3 diamonds each, or miners with bag-cutter to mine, also implying any sci/miner/rd/cmo/ai would let you clog rnd research points for this instead of important tech nodes

**Comment by ATH1909** [comment]
> > Since you're making this argument based on the Manuel ruleset, let me remind you of this cute little paragraph that 0% of the player base knows about:
> > `Only the Captain, HoS, and Warden can authorize a Permanent Sentence. Only the Captain can authorize an Execution or Forced Cyborgization.`
> > Nowhere in the server rules does it say that the warden's job is to outright kill people for trespassing into a dangerous area. This is just completely and utterly neo-player attitude garbage of having to do the least possible effort to stop someone by killing them in less than 2 seconds with a pocket combat shotgun, that not only makes combat dull but makes every situation on combat against someone with it incredibly predictable.
> 
> Hi Senjie,
> 
> I would like to remind you that the line you are quoting is not a rule, not policy, and in fact is an in-character roleplay recommendation. I believe this should be clarified as what you are saying is disingenuous.
> 
> What you quoted is from the [space law page](https://tgstation13.org/wiki/Space_Law), and I would like to remind you of this large text at the top of the page:
> 
> > If you cite Space Law in an adminhelp or anywhere outside IC communications, you will be laughed at.
> 
> Best,
> bobbah "bee" brown

based bee

Have you consdered just making the combat shotgun start the round with nonlethal ammo instead of buckshot?

**Comment by MacBlaze1** [comment]
> > > Since you're making this argument based on the Manuel ruleset, let me remind you of this cute little paragraph that 0% of the player base knows about:
> > > `Only the Captain, HoS, and Warden can authorize a Permanent Sentence. Only the Captain can authorize an Execution or Forced Cyborgization.`
> > > Nowhere in the server rules does it say that the warden's job is to outright kill people for trespassing into a dangerous area. This is just completely and utterly neo-player attitude garbage of having to do the least possible effort to stop someone by killing them in less than 2 seconds with a pocket combat shotgun, that not only makes combat dull but makes every situation on combat against someone with it incredibly predictable.
> > 
> > 
> > Hi Senjie,
> > I would like to remind you that the line you are quoting is not a rule, not policy, and in fact is an in-character roleplay recommendation. I believe this should be clarified as what you are saying is disingenuous.
> > What you quoted is from the [space law page](https://tgstation13.org/wiki/Space_Law), and I would like to remind you of this large text at the top of the page:
> > > If you cite Space Law in an adminhelp or anywhere outside IC communications, you will be laughed at.
> > 
> > 
> > Best,
> > bobbah "bee" brown
> 
> based bee
> 
> Have you consdered just making the combat shotgun start the round with nonlethal ammo instead of buckshot?

This or buffing it in some nonlethal way, or nerfing it in a lethal way? Id have to look at shotgun code again.

**Comment by HomocidlePancakes** [comment]
Well, well, well, look what the cat drag in. Let’s break down all the points from the original PR one by one: 

> The compact combat shotgun is the epitome of an overpowered unnecessary item used more for malice than good.

> Overpowered

In what way is it overpowered? Because of it’s rate of fire? You can easily shoot the normal shottie just as fast. Pressing Z isn’t that hard.

> Unnecessary 

Ok, now you are talking out of your ass. It’s given to the lad defending one of the most vital areas on the station - the armory. It makes perfect sense that he would get a slightly better weapon.

> More malice than good.

Opinions, opinions...

> The argument of a low internal magazine is moot when the weapon can instantly kill someone point blank, destroy crowds of people while being able to comfortably fit in most armor storage slots. 

> magazine

Yeah, I agree. It’s a stupid point to bring up. Having 5 shots instead of 7 barely does anything.

> kill someone point blank

This argument can be applied to all the shotguns, yet, you are removing only this one. Silly. Now, when you are fighting someone with a shottie, you should keep your distance. If you let them close, it’s entirely your fault. Not every engagement can be solved by solo bumrushing your opponent with a baton.

>Destroy crowds

Honestly, I have no idea how to respond to this point. It’s a shotgun. Makes sense it would be more effective against crowds. By the same logic we should remove explosions because they are good at dealing with crowds. Also, again, applies to all shotguns.

>Fit in slots

Same as your normal riot shottie.

> These traits have made it the perfect weapon for players who seek to use it to remove players from the game as Security rather than reduce them with non-lethal weapons.

Excuse me, what? This point is entirely unrelated to shotguns. This a player issue, and hell, if you get lethaled by sec for bad reasons, then maybe you should ahelp instead of making nerf PRs.

In short, most of the points you provided can be applied to shotguns in general, and most aren’t issues. I haven’t seen anything specific in here that applies only to the compact shottie that would warrant it’s removal. As for the contraband locker, do whatever. Honestly, one more gun doesn’t do much, but I simply don’t care about it, so I won’t bother arguing.

**Comment by Cartographer-D** [comment]
If you think this will do anything except make wardens walk 3 feet into the armory and grab a riot shotgun, you are out of your mind and need therapy.

**Comment by Cartographer-D** [comment]
image

**Comment by WhyIsCaeciliusTaken** [comment]
Looks like someone tried to tide into the armoury and got deleted by the warden.

**Comment by optimumtact** [comment]
I believe the compact combat shotgun, and all similar shotguns should be completely removed from the code.

**Comment by optimumtact** [comment]
In fact I want you to do what the original PR did only.

Remove the compact combat shotgun completely.

**Comment by tmtmtl30** [comment]
by "all similar shotguns" do you mean bulldog or combat shotgun or riot shotguns or what

**Comment by optimumtact** [comment]
Any shotgun that fits in any other slot other than the back slot is on my hit list (other than the double barrel)

**Comment by Cartographer-D** [comment]
> 
> 
> Any shotgun that fits in any other slot other than the back slot is on my hit list (other than the double barrel)

@optimumtact The compact shotgun IS only able to fit on the back slot unless I'm missing something incredible. I've tried putting it on belt and in the backpack but it does not work.

Even if it did, why not regulate it only to the back slot?

**Comment by ATH1909** [comment]
> > Any shotgun that fits in any other slot other than the back slot is on my hit list (other than the double barrel)
> 
> @optimumtact The compact shotgun IS only able to fit on the back slot unless I'm missing something incredible. I've tried putting it on belt and in the backpack but it does not work.
> 
> Even if it did, why not regulate it only to the back slot?

try the exosuit storage slot

**Comment by senjx** [comment]
contraband locker your days are counted

**Comment by Cartographer-D** [comment]
@ATH1909 is exosuit storage and suit storage not the same thing? It doesn't go on the back.
